src/common/helpers/requestMetaData/getRequestMetaData.ts

```diff
@@ -0,0 +1,71 @@
+import { UMIP } from "features/vote/helpers/fetchUMIP";
+import {
+  decodeAncillaryDataHexString,
+  getTitleFromAncillaryData,
+  getDescriptionFromAncillaryData,
+} from "./ancillaryData";
+import approvedIdentifiers from "./approvedIdentifiersTable.json";
+
+type RequestMetaData = {
+  title: string;
+  description: string;
+  umipUrl?: string;
+};
+
+/** Finds a title and description, and UMIP link (if it exists) for a proposal.
+ *
+ * There are 3 different sources of this data, depending on the proposal type:
+ *
+ * 1. For UMIPs, the title, description and UMIP link come from Contentful.
+ * 2. For requests for approved price identifiers, the title, description, and UMIP link come from the hard-coded `approvedIdentifiersTable` json file.
+ * 3. For requests from Polymarket, the title and description come from the proposal's ancillary data (note that there is no UMIP link here).
+ */
+export function getRequestMetaData(
+  ancillaryData: string,
+  proposal: string,
+  umip?: UMIP
+): RequestMetaData {
+  // if we are dealing with a UMIP, get the title, description and UMIP url from Contentful
+  const isUmip = proposal.includes("Admin");
+  if (isUmip) {
+    return {
+      title: umip?.title ?? proposal,
+      description:
+        umip?.description ?? "No description was found for this UMIP.",
+      umipUrl: umip?.umipLink,
+    };
+  }
+
+  // if we are dealing with a request for an approved price identifier, get the title, description and UMIP url from the hard-coded approvedIdentifiersTable json file
+  // we know we are dealing with a request for an approved price identifier if the proposal matches an approved identifier's title
+  const identifierDetails = approvedIdentifiers.find(
+    (id) => proposal === id.title
+  );
+  const isApprovedIdentifier = Boolean(identifierDetails);
+  if (isApprovedIdentifier) {
+    return {
+      title: identifierDetails?.title ?? proposal,
+      description:
+        identifierDetails?.summary ??
+        "No description was found for this request.",
+      umipUrl: identifierDetails?.umipLink.url,
+    };
+  }
+
+  // if the previous checks fail, we assume we are dealing with a Polymarket request
```

Comment:
Wouldn't polymarket requests match the `YES_OR_NO_QUERY` identifier and get captured in the `isApprovedIdentifier` check above?

Reply:
You're absolutely right. I missed this because I didn't know that YES_OR_NO_QUERY is both. 

I've created an array of identifiers to ignore and treat as Polymarket requests instead. We can expand it as needed with other items like this.

Do you know of others like this? Or could you point me in the direction of where to find them?

Reply:
I'm a little confused with how you're describing it -- what do you mean when you say that `YES_OR_NO_QUERY` is both?

Below is probably more information than you need! But I just wanted to detail the landscape of UMIPs, governance requests, and ancillary data in case any of that was unclear. Feel free to ignore if you already understand all of this 🙂.

The way I think about it is:
- Everything that gets requested to the DVM is either a pre-registered identifier or a governance request.
- Most requests will have _some_ ancillary data. That ancillary data _should_ be formatted according to this document: https://docs.google.com/document/d/1zhKKjgY1BupBGPPrY_WOJvui0B6DMcd-xDR8-9-SPDw/edit#heading=h.220c6fyb8tgg.
- The fields that are _required_ (if any) for a particular identifier in the ancillary data should be specified in the UMIP for the identifier (most do not require ancillary data, but this is becoming more common). Example for [YES_OR_NO_QUERY](https://github.com/UMAprotocol/UMIPs/blob/master/UMIPs/umip-107.md).
- My guess is that we're only interested in changing how we display the request for `YES_OR_NO_QUERY`, since this is the _only_ one that polymarket uses, and it's the only one with an obvious title, etc. Others have ancillary data that is really just extra info for voters, but don't have display-related info.